types/test/webpack-chain-tests.ts

```diff
@@ -0,0 +1,259 @@
+      })
+      .enforce('pre')
+      .use('babel')
+        .loader('babel-loader')
+        .options({})
+        .end()
+      .use('eslint')
+        .loader('eslint-loader')
+        .options({})
+        .after('babel')
+        .end()
+      .uses
+        .delete('babel')
+        .delete('eslint')
+        .end()
+      .pre()
+      .post()
+      .end()
+    .rules
+      .delete('compile')
+      .end()
+    .end()
+
+  .node
+    .set('__dirname', true)
+    .delete('__dirname')
+    .clear()
+    .end()
+
+  .output
+    .chunkFilename('')
+    .crossOriginLoading(true)
+    .filename('main.js')
+    .library('var')
+    .libraryTarget('var')
+    .devtoolFallbackModuleFilenameTemplate('')
+    .devtoolLineToLine('')
+    .devtoolModuleFilenameTemplate('')
+    .hashFunction('md5')
+    .hashDigest('md5')
+    .hashDigestLength(15)
+    .hashSalt('')
+    .hotUpdateChunkFilename('update')
+    .hotUpdateFunction(() => {})
+    .hotUpdateMainFilename('main')
+    .jsonpFunction('callback')
+    .path('/')
+    .pathinfo(true)
+    .publicPath('/')
+    .sourceMapFilename('index.js.map')
+    .sourcePrefix('~')
+    .strictModuleExceptionHandling(true)
+    .umdNamedDefine(true)
+    .end()
+
+  .optimization
+    .concatenateModules(true)
+    .flagIncludedChunks(true)
+    .mergeDuplicateChunks(true)
+    .minimize(true)
+    .minimizer('foo')
+      .use(webpack.DefinePlugin)
+      .end()
+    .namedChunks(true)
+    .namedModules(true)
+    .nodeEnv(true)
+    .noEmitOnErrors(true)
+    .occurrenceOrder(true)
+    .portableRecords(true)
+    .providedExports(true)
+    .removeAvailableModules(true)
+    .removeEmptyChunks(true)
+    .runtimeChunk("single")
+    .runtimeChunk({ name: ({}) => "hello" })
+    .sideEffects(true)
+    .splitChunks({})
+    .usedExports(true)
+    .end()
+
+  .performance
+    .hints(true)
+    .hints('warning')
+    .maxEntrypointSize(20000)
+    .maxAssetSize(20000)
+    .assetFilter(filename => true)
+    .end()
+
+  .resolve
+    .alias
+      .set('foo', 'bar')
+      .end()
+    .aliasFields
+      .add('foo')
+      .end()
+    .descriptionFiles
+      .add('foo')
+      .end()
+    .extensions
+      .add('.js')
+      .end()
+    .mainFields
+      .add('browser')
+      .end()
+    .mainFiles
+      .add('index.js')
+      .end()
+    .modules
+      .add('index.js')
+      .end()
+    .enforceExtension(true)
+    .enforceModuleExtension(true)
+    .unsafeCache(false)
+    .unsafeCache(/foo/)
+    .symlinks(true)
+    .cachePredicate(({ path, request }) => true)
+    .plugin('foo')
+      .use(webpack.DefinePlugin, [])
+      .end()
+    .plugins
+      .delete('foo')
+      .end()
+    .end()
+
+  .resolveLoader
+    .extensions
+      .add('.js')
+      .end()
+    .modules
+      .add('index.js')
+      .end()
+    .moduleExtensions
+      .add('.js')
+      .end()
+    .packageMains
+      .add('index.js')
+      .end()
+    .end()
+
+  .merge({})
+  .toConfig();
+  
```

Comment:
Remove extra blank line from end of file.

Reply:
ok